### PR TITLE
rpc: use minimal mode by default

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

Validators need to explicitly reduce their RPC exposure rather than deliberately increase it.  Which is a dumb default.

#### Summary of Changes

`--minimal-rpc-api` by default. (now deprecated, use ~`--extended-rpc-api`~ `--full-rpc-api` to enable the full method set)

### CODE REVIEW

File: validator/src/main.rs [hunk 1]
```diff
@@ -576,8 +576,16 @@ pub fn main() {
             Arg::with_name("minimal_rpc_api")
                 .long("--minimal-rpc-api")
                 .takes_value(false)
+                .hidden(true)
                 .help("Only expose the RPC methods required to serve snapshots to other nodes"),
         )
+        .arg(
+            Arg::with_name("extended_rpc_api")
+                .long("--extended_rpc_api")
```
Comment (mvines) [inline validator/src/main.rs:584]:
```suggestion
                .long("--extended-rpc-api")
```

File: validator/src/main.rs [hunk 2]
```diff
@@ -576,8 +576,16 @@ pub fn main() {
             Arg::with_name("minimal_rpc_api")
                 .long("--minimal-rpc-api")
                 .takes_value(false)
+                .hidden(true)
                 .help("Only expose the RPC methods required to serve snapshots to other nodes"),
         )
+        .arg(
+            Arg::with_name("extended_rpc_api")
```
Comment (mvines) [inline validator/src/main.rs:583]: Code looks great, I just want to bikeshed the name a little.  I feel `--full-rpc-api` is more accurate.  Alternatively `--complete-rpc-api`?

Comment (CriesofCarrots) [reply validator/src/main.rs:583]: I also prefer something like this ^
I've no preference between "full" and "complete"

Comment (t-nelson) [reply validator/src/main.rs:583]: Yeah I was wrestling with whether we should commit to this set being "full/complete" given the deprecated method set.

I'd additionally considered something like `--enable-rpc-method-set`, which takes a CSV following the available traits `[minimal*, bank, accounts, deprecated-v1.9, deprecated-v1.7, full, obsolete-v17]`? Then we additionally hide the `--enable-rpc-obsolete_v1_7` flag

Comment (mvines) [reply validator/src/main.rs:583]: `--full-rpc-api` feels like a nice complement to `--minimal-rpc-api` to me.

I'm not opposed to something like `--rpc-method-set` but also not sure it buys us too much either. The common options seem to be "minimal", "full", or "full" with "enable-rpc-obsolete_v1_7".  It's unclear to me who would want more granular control over the exposed methods

Comment (t-nelson) [reply validator/src/main.rs:583]: > but also not sure it buys us too much either

Well surely we have the additional complexity in the code base for a reason, right? :wink: 

I'll go with `--full-rpc-api` from now

**Review by mvines — APPROVED** [review]

**Review by CriesofCarrots — COMMENTED** [review]
CI failures look legit. `TestValidator` needs attention?

**Comment by mvines** [comment]
ah, `TestValidator` is prolly getting the minimal api as well ☠️

**Comment by t-nelson** [comment]
`grep` fails me again!  I'll fix it up on Monday

**Comment by t-nelson** [comment]
Gonna merge on red and follow up with a fix to appease Travis CI's no-output timeout, which I don't believe is related to these changes

**Pull request merged**